r/vignettes/install.Rmd

```diff
@@ -0,0 +1,251 @@
+which introduces all sorts of additional ways for things to go wrong.
+
+Our goal is to make `install.packages("arrow")` "just work" for as many Linux distributions,
+versions, and configurations as possible.
+This document describes how it works and the options for fine-tuning Linux installation.
+The intended audience for this document is `arrow` R package users on Linux, not developers.
+If you're contributing to the Arrow project,
+you'll probably want to manage your C++ installation more directly.
+Note also that if you use `conda` to manage your R environment, this document does not apply.
+You can `conda install -c conda-forge r-arrow` and you'll get the latest official
+release of the R package along with any C++ dependencies.
+
+# Installation basics
+
+Install the latest release of `arrow` from CRAN with
+
+```r
+install.packages("arrow")
+```
+
+Daily development builds, which are not official releases,
+can be installed from the Ursa Labs repository:
+
+```r
+install.packages("arrow", repos = "https://dl.bintray.com/ursalabs/arrow-r")
+```
+
+There currently are no daily `conda` builds.
+
+You can also install the R package from a git checkout:
+
+```shell
+git clone https://github.com/apache/arrow
+cd arrow/r
+R CMD INSTALL .
+```
+
+<!-- TODO: does remotes::install_github("apache/arrow/r") work now? -->
+
+# How dependencies are resolved
+
+In order for the `arrow` R package to work, it needs the Arrow C++ library.
+There are a number of ways you can get it: a system package; a library you've
+built yourself outside of the context of installing the R package;
+or, if you don't already have it, the R package will attempt to resolve it
+automatically when it installs. This section outlines the logic of how that
+dependency resolution occurs.
+
+When you install the `arrow` R package on Linux,
+it will first attempt to find the Arrow C++ libraries on your system using
+the `pkg-config` command.
+This will find either installed system packages or libraries you've built yourself.
+
+If you are authorized to install system packages, and you're installing a CRAN release,
+you may want to use the official Apache Arrow release packages corresponding to the R package version.
+See the [Arrow project installation page](https://arrow.apache.org/install/)
+to find pre-compiled binary packages for some common Linux distributions,
+including Debian, Ubuntu, and CentOS.
+You'll need to install `libparquet-dev` on Debian and Ubuntu, or `parquet-devel` on CentOS.
+This will also automatically install the Arrow C++ library as a dependency.
+In order for `install.packages("arrow")` to work with these system packages,
+you'll need to install them before installing the R package.
+
+Alternatively, if you've built the Arrow C++ libraries locally from source
+but haven't installed them where `pkg-config` will find them,
+you can instead set `INCLUDE_DIR` and `LIB_DIR` environment variables to point to their location.
```

Comment:
You might also want to mention `PKG_CONFIG_PATH` for when the user builds arrow locally but installs with custom CMAKE_INSTALL_PREFIX.

Reply:
I'm not familiar with that variable. Could you suggest some language to include?

Reply:
How about this?

Before:

> you can instead set `INCLUDE_DIR` and `LIB_DIR` environment variables to point to their location.

After:

> you can set the `PKG_CONFIG_PATH=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig` environment variable such as `PKG_CONFIG_PATH=/home/user/local/lib/pkgconfig` to find the Arrow C++ libraries. If you can't set `PKG_CONFIG_PATH`,  you can instead set `INCLUDE_DIR` and `LIB_DIR` environment variables to point to their location.
```